File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_e4/type2_renamed_1_substring_search.py
def find_pattern(text, pattern):
    if not pattern: return 0
    
    for index in range(len(text) - len(pattern) + 1):
        for offset in range(len(pattern)):
            if text[index + offset] != pattern[offset]:
                break
            if offset == len(pattern) - 1:
                return index
                
    return -1

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_e4/type2_renamed_1_substring_search.java
public class Solution {
    public int find_pattern(String text, String pattern) {
        if (pattern == null || pattern.length() == 0) return 0;
        
        for (int index = 0; index <= text.length() - pattern.length(); index++) {
            for (int offset = 0; offset < pattern.length(); offset++) {
                if (text.charAt(index + offset) != pattern.charAt(offset)) {
                    break;
                }
                if (offset == pattern.length() - 1) {
                    return index;
                }
            }
        }
        return -1;
    }
}
